src/mito/src/engine.rs

```diff
@@ -381,11 +381,181 @@ fn validate_create_table_request(request: &CreateTableRequest) -> Result<()> {
+        let table_ref = TableReference {
+            catalog: catalog_name,
+            schema: schema_name,
+            table: table_name,
+        };
+
+        let table_id = request.table_id;
+        let engine_ctx = StorageEngineContext::default();
+        let table_dir = table_dir(catalog_name, schema_name, table_id);
+
+        let Some((manifest, table_info)) = self
+                            .recover_table_manifest_and_info(table_name, &table_dir)
+                            .await.map_err(BoxedError::new)
+                            .context(table_error::TableOperationSnafu)? else { return Ok(None) };
+
+        let opts = OpenOptions {
+            parent_dir: table_dir.to_string(),
+            write_buffer_size: table_info
+                .meta
+                .options
+                .write_buffer_size
+                .map(|s| s.0 as usize),
+            ttl: table_info.meta.options.ttl,
+            compaction_time_window: table_info.meta.options.compaction_time_window,
+        };
+
+        debug!(
+            "Opening table {}, table info recovered: {:?}",
+            table_id, table_info
+        );
+
+        for target_region in &request.region_numbers {
+            if !table_info.meta.region_numbers.contains(target_region) {
+                table_error::RegionNotFoundSnafu {
+                    table: table_ref.to_string(),
+                    region: *target_region,
+                }
+                .fail()?
+            }
+        }
+
+        let mut regions = HashMap::with_capacity(table_info.meta.region_numbers.len());
+
+        for region_number in &request.region_numbers {
+            let region = self
+                .open_region(&engine_ctx, table_id, *region_number, &table_ref, &opts)
+                .await?;
+            regions.insert(*region_number, region);
+        }
+
+        let table = Arc::new(MitoTable::new(table_info, regions, manifest));
+
+        Ok(Some(table))
+    }
+
+    async fn open_region(
+        &self,
+        engine_ctx: &StorageEngineContext,
+        table_id: TableId,
+        region_number: RegionNumber,
+        table_ref: &TableReference<'_>,
+        opts: &OpenOptions,
+    ) -> TableResult<S::Region> {
+        let region_name = region_name(table_id, region_number);
+        let region = self
+            .storage_engine
+            .open_region(engine_ctx, &region_name, opts)
+            .await
+            .map_err(BoxedError::new)
+            .context(table_error::TableOperationSnafu)?
+            .with_context(|| RegionNotFoundSnafu {
+                table: format!(
+                    "{}.{}.{}",
+                    table_ref.catalog, table_ref.schema, table_ref.table
+                ),
+                region: region_number,
+            })
+            .map_err(BoxedError::new)
+            .context(table_error::TableOperationSnafu)?;
+
+        Ok(region)
+    }
+
+    /// Loads regions
+    async fn load_regions(
```

Comment:
suggestion: Consider renaming to `load_missing_regions`
```suggestion
    async fn load_missing_regions(
```